src/lua/LuaSystemBody.cpp

```diff
@@ -148,15 +148,16 @@ static int l_sbody_attr_parent(lua_State *l)
 {
 	SystemBody *sbody = LuaObject<SystemBody>::CheckFromLua(1);
 
-	// sbody->parent is 0 as it was cleared by the acquirer. we need to go
-	// back to the starsystem proper to get what we need.
-	RefCountedPtr<StarSystem> s = Pi::game->GetGalaxy()->GetStarSystem(sbody->GetPath());
-	SystemBody *live_sbody = s->GetBodyByPath(sbody->GetPath());
-
-	if (!live_sbody->GetParent())
+	if (!sbody->GetStarSystem()) {
+		// orphan, its system has already been deleted, but SystemPath remains
+		// we'll make a new one just like it
+		RefCountedPtr<StarSystem> s = Pi::game->GetGalaxy()->GetStarSystem(sbody->GetPath());
+		sbody = s->GetBodyByPath(sbody->GetPath());
```

Comment:
Note: `GetStarSystem` does not maintain a strong reference to the returned StarSystem (just a weak one so it can be removed from the master cache when all other references expire), so this system (AFAIK) will be deleted as soon as the `if` block scope is closed. This would orphan the just-created system body, leaving it still without parent.

If this is behaving differently than I've described, something's holding on to a reference to the star system that's been created and I retract my comment.

Reply:
Yes, you're right! Probably nothing broke for me, because in fact I never managed to get into this branch when testing the game. Apparently, it is enough for the system to survive until the end of the function.

Reply:
In general I think the code pattern of interacting with orphaned SystemBody objects should be strongly discouraged, due to the steep cost of re-generating the system on every access to the `parent` member of both the current orphaned body and the returned parent body. However, this is an acceptable solution to prevent total breakage.